PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/AdlsGen2VirtualFile.java

```
@@ -0,0 +1,124 @@
+    private String path;
+    private VirtualFileSystem fileSystem;
+    private boolean isDirectory;
+    private String prefix;
+    private List<VirtualFile> children = new ArrayList<>();
+    private VirtualFile parent;
+    public static VirtualFile empty = new AdlsGen2VirtualFile("", false, "", new HttpFileSystemImpl());
+
+    public AdlsGen2VirtualFile(String path, boolean isDirectory, String prefix, VirtualFileSystem fileSystem) {
+        this.name = path.substring(path.lastIndexOf("/") + 1);
+        this.path = path;
+        this.isDirectory = isDirectory;
+        this.prefix = prefix;
+        this.fileSystem = fileSystem;
+    }
+
+    @Nullable
+    @Override
+    public RemoteFileInfo getFileInfo() {
+        return null;
+    }
+
+    @NotNull
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @NotNull
+    @Override
+    public VirtualFileSystem getFileSystem() {
+        return this.fileSystem;
+    }
+
+    @NotNull
+    @Override
+    public String getPath() {
+        return prefix != null ? String.format("%s/%s", this.prefix, this.path) : this.path;
+    }
+
+    @Override
+    public boolean isWritable() {
+        return false;
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return this.isDirectory;
+    }
+
+    @Override
+    public boolean isValid() {
+        return true;
+    }
+
+    @Override
+    @Nullable
+    public VirtualFile getParent() {
+        return this.parent;
+    }
+
+    public void setParent(VirtualFile parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public VirtualFile[] getChildren() {
+        return this.children.toArray(new AdlsGen2VirtualFile[0]);
+    }
+
+    public void addChildren(AdlsGen2VirtualFile vf) {
+        this.children.add(vf);
+    }
+
+    @NotNull
+    @Override
+    public OutputStream getOutputStream(Object requestor, long newModificationStamp, long newTimeStamp) throws IOException {
+        return null;
```

Returns a null to `@NotNull` function

---

You can `throw new UnsupportedOperationException("ADLS Gen2 FileSystem is read-only");`